src/components/Form/types.ts

```diff
@@ -0,0 +1,66 @@
+import type {FocusEvent, MutableRefObject, ReactNode} from 'react';
+import type {StyleProp, ViewStyle} from 'react-native';
+import type TextInput from '@components/TextInput';
+import type {BaseTextInputRef} from '@components/TextInput/BaseTextInput/types';
+import type {OnyxFormKey, OnyxValues} from '@src/ONYXKEYS';
+import type {Form} from '@src/types/onyx';
+
+type ValueType = 'string' | 'boolean' | 'date';
+
+type ValidInput = typeof TextInput;
+
+type InputProps<TInput extends ValidInput> = Parameters<TInput>[0] & {
+    shouldSetTouchedOnBlurOnly?: boolean;
+    onValueChange?: (value: unknown, key: string) => void;
+    onTouched?: (event: unknown) => void;
+    valueType?: ValueType;
+    onBlur: (event: FocusEvent | Parameters<NonNullable<Parameters<TInput>[0]['onBlur']>>[0]) => void;
+};
```

Comment:
Maybe we could leave some comment explaining that this type is doing? For example, why `Parameters<TInput>[0]`?

Reply:
I see now the purpose, what about trying this solution I did in the past for `InputWrapper`? https://github.com/Expensify/App/compare/main...fabioh8010:expensify-app:ts/component/InputWrapper

Reply:
I'm not sure I understand what you mean.